### PR TITLE
Re-enable retrieval of model alias per controller.

### PR DESCRIPTION
**Pull request by dereuromark**
When upgrading a plugin one faces the challenge of accessing the now protected class property.
In this case $modelClass and the resulting alias per controller

```php
// Inside component
$modelName = $this->Controller->modelClass;
		
list(, $modelName) = pluginSplit($modelName);
$this->setConfig('modelName', $modelName);
if (!$this->Controller->{$modelName}->behaviors()->has('...')) {
	$this->Controller->{$modelName}->behaviors()->load('...', $this->_config);
}
```

With this suggested change this would be able again
```php
$modelName = $this->Controller->getModelAlias();

$this->setConfig('modelName', $modelName);
if (!$this->Controller->{$modelName}->behaviors()->has('...')) {
	$this->Controller->{$modelName}->behaviors()->load('...', $this->_config);
}
```
Or other generic operations done by a generic component assisting:
```php
$modelName = $this->Controller->getModelAlias();
if ($this->Controller->{$modelName}->do()) {
	...
}
```

Or are there better alternatives?

Force reading the field with reflection is an option but sounds like a less clean approach.

### CODE REVIEW

**Review by jadb — APPROVED** [review]

**Comment by ADmad** [comment]
This those target `4.next`. We no longer add new feature or change public API (even if it's BC) in patch releases.

File: src/Datasource/ModelAwareTrait.php [hunk 1]
```diff
@@ -73,52 +73,81 @@ protected function _setModelClass(string $name): void
...
-     *  The name can be an alias like `'Post'` or FQCN like `App\Model\Table\PostsTable::class`.
      * @param string|null $modelType The type of repository to load. Defaults to the modelType() value.
-     * @return \Cake\Datasource\RepositoryInterface The model instance created.
-     * @throws \Cake\Datasource\Exception\MissingModelException If the model class cannot be found.
      * @throws \UnexpectedValueException If $modelClass argument is not provided
      *   and ModelAwareTrait::$modelClass property value is empty.
+     * @return string
      */
-    public function loadModel(?string $modelClass = null, ?string $modelType = null): RepositoryInterface
+    public function getModelAlias(?string $modelClass = null, ?string $modelType = null): string
```
Comment (dereuromark) [inline src/Datasource/ModelAwareTrait.php]: Or should this be named `getModelName()` ?

File: src/Datasource/ModelAwareTrait.php [hunk 2]
```diff
@@ -73,52 +73,81 @@ protected function _setModelClass(string $name): void
...
 
     /**
-     * Loads and constructs repository objects required by this object
-     *
-     * Typically used to load ORM Table objects as required. Can
-     * also be used to load other types of repository objects your application uses.
-     *
-     * If a repository provider does not return an object a MissingModelException will
-     * be thrown.
-     *
```
Comment (markstory) [inline src/Datasource/ModelAwareTrait.php]: Shouldn't this method have a description?

**Comment by markstory** [comment]
Closing in favour of #14356

**Pull request closed without merging**